src/redis/redis.client.ts

```diff
@@ -39,4 +43,24 @@ export class RedisClient implements OnModuleDestroy {
   ): Promise<void> {
     await this.redisClient.set(`${prefix}:${key}`, value, 'EX', expiry)
   }
+
+  async acquireLock(key: string, value: string): Promise<boolean> {
+    const lockExpiryInSeconds = 30
+    // NX option is used to set the key only if it does not already exist else it will return null
+    const result = await this.redisClient.set(
+      key,
+      value,
+      'EX',
```

Comment:
Et si on en faisait une variable nommée qui explique un peu à quoi ça sert ?

Reply:
clean code it is